app/lib/meadow/search/index.ex

```diff
@@ -52,6 +52,29 @@ defmodule Meadow.Search.Index do
       ) do
     pipeline = %{
       "description" => "Search pipeline for #{name}",
+      "request_processors" => [
+        %{
+          "filter_query" => %{
+            "description" => "Restricts requests to publicly visible documents",
+            "query" => %{
+              "bool" => %{
+                "should" => [
+                  %{
+                    "term" => %{
+                      "visibility" => "public"
+                    }
+                  },
+                  %{
+                    "term" => %{
+                      "visibility" => "authenticated"
+                    }
+                  }
+                ]
+              }
+            }
```

Comment:
I think a single `terms` query will outperform two combined `term` queries. (and be easier to read)

```suggestion
            "query" => %{
              "terms" => {
                "visibility": ["public", "authenticated"]
              }
            }
```